File: oolab/src/main/java/agh/ics/oop/model/MapDirection.java
package agh.ics.oop.model;

public enum MapDirection {
    NORTH(1),EAST(2) ,SOUTH(3), WEST(4) ;
    private int id;

    MapDirection(int id){
        this.id=id;
    }

    public String toString(){
        return switch(this){
            case NORTH -> "Północ";
            case SOUTH -> "Południe";
            case WEST -> "Zachód";
            case EAST -> "Wschód";
        };
    }

    public MapDirection next(){
        MapDirection[] dir=MapDirection.values();
        int i=0;
        for (; dir[i]!=(this);i++){

        }
        i++;
        i%=dir.length;
        return dir[i];

    }

    public MapDirection previous(){
        MapDirection[] dir=MapDirection.values();
        int i=0;
        for (; dir[i]!=(this);i++){

        }
        i+=dir.length-1;
        i%=dir.length;
        return dir[i];
    }

    public Vector2D toUnitVector(){
        return switch(this){
            case NORTH -> new Vector2D(0,1);
            case SOUTH ->  new Vector2D(0,-1);
            case WEST -> new Vector2D(-1,0);
            case EAST -> new Vector2D(1,0);
        };
    }
}

File: kotlin_lab/untitled/src/main/kotlin/model/kotlin/MapDirection.kt
package model.kotlin

import model.kotlin.Vector2d


enum class MapDirection(i: Int) {
    NORTH(1),
    EAST(2),
    SOUTH(3),
    WEST(4);

    override fun toString(): String = when (this) {
        NORTH -> "N";
        SOUTH -> "S";
        WEST -> "W";
        EAST -> "E";
    }

    operator fun next() = entries.toTypedArray().elementAt(
        (
                this.ordinal + 1) % MapDirection.entries.size
    )

    fun previous() = entries.toTypedArray().elementAt(
        (
                this.ordinal - 1 + MapDirection.entries.size) % MapDirection.entries.size
    )


}